common/components/componentsBySection/CreateEventProject/CreateEventProjectSelect.jsx

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import React from "react";
+import DjangoCSRFToken from "django-react-csrftoken";
+import { Container } from "flux/utils";
+import _ from "lodash";
+import FormValidation from "../../../components/forms/FormValidation.jsx";
+import type { Validator } from "../../forms/FormValidation.jsx";
+import { FormPropsBase, FormStateBase } from "../../utils/forms.js";
+import UniversalDispatcher from "../../stores/UniversalDispatcher.js";
+import type { FormFieldValidator } from "../../utils/validation.js";
+import FormFieldsStore from "../../stores/FormFieldsStore.js";
+import type { EventProjectAPIDetails } from "../../utils/EventProjectAPIUtils.js";
+import InlineFormError from "../../forms/InlineFormError.jsx";
+import CurrentUser, {
+  UserContext,
+  MyProjectData,
+} from "../../utils/CurrentUser.js";
+import FormSelector from "../../common/selection/FormSelector.jsx";
+import HiddenFormFields from "../../forms/HiddenFormFields.jsx";
+import url from "../../utils/url.js";
+
+type FormFields = {|
+  project_id: number,
+|};
+
+type Props = {|
+  project: ?EventProjectAPIDetails,
+  readyForSubmit: () => () => boolean,
+|} & FormPropsBase<FormFields>;
+
+type State = {|
+  owned_projects: $ReadOnlyArray<MyProjectData>,
+  isAlreadySelected: boolean,
+  formIsValid: boolean,
+  validations: $ReadOnlyArray<Validator>,
+|} & FormStateBase<FormFields>;
+
+/**
+ * Encapsulates form for selecting project to RSVP for event
+ */
```

Comment:
I've seen a lot of these and I'm just commenting to say that I like seeing them. A little inline help on what a component does is excellent practice. 👍